README.md

```diff
@@ -1,8 +1,380 @@
-## zanzibar [![GoDoc][doc-img]][doc] [![Build Status][ci-img]][ci] [![Coverage Status][cov-img]][cov] [![Go Report Card][go-report-img]][go-report]
+# <img src="zanzibar.png" width="352">    [![GoDoc][doc-img]][doc] [![Build Status][ci-img]][ci] [![Coverage Status][cov-img]][cov] [![Go Report Card][go-report-img]][go-report]
 
-A build system & runtime component to generate configuration driven gateways.
 
-## Installation
+Zanzibar is an extensible framework to build configuration driven web applications. The goal of Zanizibar is to simplify application development into two steps:
+
+1. write configurations for the application and its components;
+2. write code to implement and test business logic.
+
+Based on the configurations, Zanizbar generates boilerplates and glue code, wires them up with your business domain code and the runtime components Zanzibar provides to create a deployable binary.
+
+The builtin components of Zanzibar makes it easy to develop mircoserivces and gateway services that proxy or orchestrate microsevices. It is also simple to extend Zanzibar with custom plugins to ease the development of applications that suit your specific needs.
+
+## Concepts
+Zanzibar is built on three pillars: module, config, code generation.
+### Module
+Modules are the components that a Zanzibar application is made of. A module belongs to a `ModuleClass`, has a `type` and can have dependencies on other modules.
+
+#### ModuleClass
+`ModuleClass` abstracts the functionality of a specific class of components. Zanzibar predefines a few module classes, i.e., `client`, `endpoint`, `middleware` and `service`. Each represents a corresponding abstraction:
+
+ModuleClass | Abstraction
+----------- | -----------
+client | clients to communicate with downstreams, e.g., database clients and RPC clients
+endpoint | application interfaces exposed to upstreams
+middleware | common functionality that has less to do with business logic, e.g., rate limiting middleware
+servcie | high level appliation abstraction, e.g., a demo service that prints "Hello World!"
+
+#### Type
+The module `type` differentiates module instances of the same `ModuleClass` with further classification.
```

Comment:
You should have a subsection here for each module type if you are going to enumerate all of the module types for each module class for more structure.